### ISSUE TITLE
Bugs and crashes on moving windows between workspaces with different layouts.

### ISSUE THREAD

**Issue by ArkadyBuryakov** [zakk4223/hyprWorkspaceLayouts#5]
# Description
I have two workspaces with different layouts:
№ 1 - with dwindle layout (assigned)
№ 2 - with master layout (default)

Hypland version: 0.34.0.r61.754eaf5b-1

Moving windows between those workspaces leads to following bugs:

## Phantom window after moving to a different layout workspace
**Steps to reproduce**
1. Create windows on workspace 1
2. Move a window to workspace 2
3. Return focus to workspace 1

**Result:** There is a blank space where the window previously was. Tiling works as there is a phantom window.

## Crash on return to workspace after closing moved window
**Steps to reproduce**
1. Create windows on workspace 1
2. Move a window to workspace 2
3. Close the window
4. Return to workspace 1

**Result:** Hyprland crashes.

**Comment by VirtCode** [comment]
Can reproduce. Happens when using keyboard binds to move the windows, not with mouse binds.
I may have a look at it on Wednesday, if I have time.

**Comment by zakk4223** [comment]
Oh hmm, this is annoying. 

https://github.com/hyprwm/Hyprland/blob/91e8c428431deac1e5eb8e537f002ab960777174/src/Compositor.cpp#L2695

here hyprland calls onWindowRemovedTiling, then sets the workspace id etc. Then calls onWindowCreatedTiling

The problem is this line: https://github.com/hyprwm/Hyprland/blob/91e8c428431deac1e5eb8e537f002ab960777174/src/Compositor.cpp#L2689

that function sets the workspace id of the window too. So when onWindowRemovedTiling is called, the workspaceID is already set to the new workspace. So the plugin uses the 'new' workspace for the proxied remove call.

I think I'm just going to have to keep a global state of all window->workspaceid mappings in the plugin and use that to lookup the layout for each operation.

**Comment by zakk4223** [comment]
Try this commit, please

**Comment by VirtCode** [comment]
I did some further testing and it seems to work fine now when switching between ordinary workspaces with different layouts. However, it doesn't seem to work correctly with special workspaces yet. When moving a window to a special one, it doesn't tile and stays the same size.

It's quite easy to reproduce:
I have `master` as a default layout and set the layout of a special workspace to `dwindle`. 
1. Create at least 2 windows on a normal `master` workspace.
2. Move a window to your the special workspace with a keybind -> the window won't tile.
3. Move a couple of more windows to the special workspace -> they'll start to overlap etc.

When opening a new window on the special workspace all windows are tiled correctly again.

I also noticed that opening a special workspace on another monitor does also not show the window (probably because they are not retiled) until I open a new one.

**Comment by zakk4223** [comment]
special workspaces are weird in hyprland.... please try latest commit

**Comment by VirtCode** [comment]
This fixes all the issues I've had with the special workspaces, thanks!
However, I've found another one. It took me a lot of time to find how to reproduce the issue but this should work:

I have two monitors with differing layouts. I've added some numbers for clarity:
1. Open two windows on one monitor (1).
2. Move one of them to the other monitor (2) **via keybind**.
3. Move it back **with the mouse** (to 1).
4. Move it again to the the other monitor (2) **via keybind**. -> The monitor (1) does not retile.
5. Focus the first monitor (1) with the mouse. -> The window disappears from the other monitor (2).
6. Focus the other monitor (2) with the mouse. -> The window appears again. etc.

Sometimes the compositor also crashes when closing the windows after reproducing it, but I couldn't reproduce that reliably. I could also reproduce it with your latest commits (from ~1-2 hours ago).
